front/src/applications/rollingStockEditor/components/RollingStockEditorForm.tsx

```diff
@@ -165,144 +73,200 @@ const RollingStockEditorForm = ({
-            default_curve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
-          },
+  const defaultValues: RollingStockParametersValues = {
+    railjsonVersion: rollingStockData?.railjson_version || '',
+    name: rollingStockData?.name || '',
+    detail: rollingStockData?.metadata.detail || '',
+    family: rollingStockData?.metadata.family || '',
+    grouping: rollingStockData?.metadata.grouping || '',
+    number: rollingStockData?.metadata.number || '',
+    reference: rollingStockData?.metadata.reference || '',
+    series: rollingStockData?.metadata.series || '',
+    subseries: rollingStockData?.metadata.subseries || '',
+    type: rollingStockData?.metadata.type || '',
+    unit: rollingStockData?.metadata.unit || '',
+    length: rollingStockData?.length || 0,
+    mass: rollingStockData ? rollingStockData.mass / 1000 : 0, // The mass received is in kg and should appear in tons.
+    maxSpeed: rollingStockData ? rollingStockData.max_speed * 3.6 : 0, // The speed received is in m/s and should appear in km/h.
+    startupTime: rollingStockData?.startup_time || 0,
+    startupAcceleration: rollingStockData?.startup_acceleration || 0,
+    comfortAcceleration: rollingStockData?.comfort_acceleration || 0.01,
+    gammaValue: rollingStockData?.gamma.value || 0.01,
+    inertiaCoefficient: rollingStockData?.inertia_coefficient || 1,
+    loadingGauge: rollingStockData?.loading_gauge || 'G1',
+    rollingResistanceA: rollingStockData?.rolling_resistance.A || 0,
+    rollingResistanceB: rollingStockData?.rolling_resistance.B || 0,
+    rollingResistanceC: rollingStockData?.rolling_resistance.C || 0,
+    electricalPowerStartupTime: rollingStockData?.electrical_power_startup_time || null,
+    raisePantographTime: rollingStockData?.raise_pantograph_time || null,
+    defaultMode: rollingStockData?.effort_curves.default_mode || selectedMode,
+    effortCurves: {
+      modes: {
+        [`${selectedMode}`]: {
+          curves: defaultRollingstockMode.modes[`${selectedMode}`].curves,
+          isElectric: defaultRollingstockMode.modes[`${selectedMode}`].is_electric,
+          defaultCurve: defaultRollingstockMode.modes[`${selectedMode}`].default_curve,
         },
       },
     },
-  } as UseFormProps<RollingStockParametersValues>);
+  };
+
+  const [rollingStockValues, setRollingStockValues] =
+    useState<RollingStockParametersValues>(defaultValues);
 
-  // in the top of the form we display all the metadatas
-  // the array is split in two to display the metadatas in two columns
-  const metadataForm = (position: boolean, registerForm: UseFormRegister<FieldValues>) =>
-    Object.keys(schema.properties)
-      .filter(
-        (property, index) =>
-          property === Metadata[property as Metadata] && (position ? index <= 5 : index > 5)
-      )
+  const metadataForm = (data: SchemaProperty[]) =>
+    data
+      .filter((property) => property.title === Metadata[property.title as Metadata])
```

Comment:
Transform Metadata in an array and use includes. You can also rename it RollingStockMetadataFields
```suggestion
      .filter((property) => RollingStockMetadataFields.includes(property.title))
```

Reply:
Done